nautobot/core/celery/__init__.py

```diff
@@ -217,5 +217,6 @@ def _loads(obj):
 def register_jobs(*jobs):
     """Helper method to register jobs with Celery"""
     for job in jobs:
+        # TODO: should we only register a job if it corresponds to a Job database record?
```

Comment:
Don't we use the celery registry to create the database entries? We don't want to go back to trying to find jobs manually, right?

Reply:
Yeah, it might get circular if we try to do this. I'm just not a big fan of the current behavior where we register it into Celery even if it failed validation when we tried to create the database entry.

Reply:
We could prune the celery registry based on the database models but it would get complicated on the worker(s) since they don't know when the database is ready